runtime/interpreter/interpreter.go

```diff
@@ -3817,8 +3837,37 @@ func (interpreter *Interpreter) checkTypeLoading(staticType StaticType) (typeErr
 		}
 	}()
 
-	// Here it is only interested in whether the type can be properly loaded.
-	_, typeError = interpreter.ConvertStaticToSemaType(staticType)
```

Comment:
**Fix 1**: Here, it is not sufficient to just load the types. But rather, need to follow links, similar to `pathCapabilityCheckFunction ` function, and check the type of the actual value stored.